### PR TITLE
Unwrap optional once

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/SecretManagerConfigurationClient.java [hunk 1]
```diff
@@ -105,14 +108,15 @@ private Map<Integer, String> configCandidates(Environment environment) {
         int priority = EnvironmentPropertySource.POSITION + 150;
 
         if (configurationProperties.isDefaultConfigEnabled()) {
-            Optional<String> applicationName = environment.getProperty("micronaut.application.name", String.class);
-            candidates.put(EnvironmentPropertySource.POSITION + 101, "application");
-            applicationName.ifPresent(s -> candidates.put(EnvironmentPropertySource.POSITION + 102, s));
-
+            String applicationName = environment.getProperty(MICRONAUT_APPLICATION_NAME, String.class).orElse(null);
```
Comment (guillermocalvo) [inline gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/SecretManagerConfigurationClient.java:111]: @sdelamo I'd like to know what's your motivation to replace optionals with nullable references? Performance? Readability?

**Review by guillermocalvo — APPROVED** [review]
I'm not a big fan of replacing optionals with null references, but I guess I could live with these changes if you feel strongly about it.

Then again this PR makes me wonder why `environment.getProperty` returns an `Optional` in the first place if we (as a framework) don't want to touch them. 🤔

**Comment by sdelamo** [comment]
> I'm not a big fan of replacing optionals with null references, but I guess I could live with these changes if you feel strongly about it.
> 
> Then again this PR makes me wonder why `environment.getProperty` returns an `Optional` in the first place if we (as a framework) don't want to touch them. 🤔

That does not reflect the framework views. We don't have anything against optionals.

File: gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/SecretManagerConfigurationClient.java (hunk 1 above)
Comment (guillermocalvo) [reply gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/SecretManagerConfigurationClient.java:111]: I find this type of optimizations kind of premature, but I admit null checks will be somewhat faster in most scenarios.

**Pull request merged**